sled-agent/src/sim/server.rs

```diff
@@ -32,36 +32,30 @@ use uuid::Uuid;
 /// Packages up a [`SledAgent`], running the sled agent API under a Dropshot
 /// server wired up to the sled agent
 pub struct Server {
+    // Configuration used to start server
+    config: Config,
+    log: Logger,
+
     /// underlying sled agent
     pub sled_agent: Arc<SledAgent>,
     /// dropshot server for the API
     pub http_server: dropshot::HttpServer<Arc<SledAgent>>,
     /// simulated pantry server
-    pub pantry_server: PantryServer,
-    /// real internal dns server storage dir
-    pub dns_server_storage_dir: tempfile::TempDir,
-    /// real internal dns server
-    pub dns_server: dns_server::dns_server::ServerHandle,
-    /// real internal dns dropshot server
-    pub dns_dropshot_server:
-        dropshot::HttpServer<dns_server::http_server::Context>,
```

Comment:
This is kinda a heavy thing to force on every single Sled Agent server. It should be possible to run simulated sled agents without DNS servers, so this has been decoupled.